libfoundation/include/foundation-auto.h

```diff
@@ -87,9 +87,14 @@ template<typename T> class MCAutoValueRefBase
         m_value = (value) ? (T)MCValueRetain(value) : NULL;
     }
     
-    inline T& operator ! (void)
+    inline operator bool ()
     {
-		return m_value;
+        return m_value != nil;
+    }
+    
+    inline bool operator ! ()
+    {
+		return m_value == nil;
```

Comment:
@livecodefraser I don't think we want to do this, because it introduces confusion when applied to an `MCAutoBooleanRef`.  For now, could you please leave `operator bool()` and `operator !()` undefined for `MCAutoValueRefBase` and add an `IsSet()` method (or equivalent) instead?